packages/autotask-client/src/bin.ts

```diff
@@ -1,41 +1,159 @@
 #!/usr/bin/env node
+import { AxiosError } from 'axios';
 import 'dotenv/config';
-import { existsSync } from 'fs';
+import { createWriteStream, WriteStream } from 'fs';
 import { argv } from 'process';
-import { AutotaskClient, VERSION } from '.';
+import { VERSION } from '.';
+import { AutotaskRunListItemResponse } from './models/autotask-run.res';
+import { initClient, output, validateId, validatePath } from './utils';
+
+type Command = 'update-code' | 'tail-runs' | 'execute-run';
 
 function printUsage() {
   console.error(`Defender Autotask Client (version ${VERSION})\n`);
   console.error('Usage: defender-autotask update-code $AUTOTASK_ID $PATH');
   console.error('\nExample:\n  defender-autotask update-code 19ef0257-bba4-4723-a18f-67d96726213e ./lib/autotask\n');
+  console.error('Usage: defender-autotask tail-runs $AUTOTASK_ID');
+  console.error('\nExample:\n  defender-autotask tail-runs 19ef0257-bba4-4723-a18f-67d96726213e\n');
+  console.error('Usage: defender-autotask execute-run $AUTOTASK_ID');
+  console.error('\nExample:\n  defender-autotask execute-run 19ef0257-bba4-4723-a18f-67d96726213e\n');
+}
+
+/**
+ * Makes sure that mandatory params for the given command are present.
+ * @param command The command to validate.
+ */
+function mandatoryParamGuard(command: Command) {
+  switch (command) {
+    case 'update-code':
+      if (!argv[3] || !argv[4]) {
+        printUsage();
+        process.exit(1);
+      }
+      break;
+    // Same requirements for now
+    case 'tail-runs':
+    case 'execute-run':
+      if (!argv[3]) {
+        printUsage();
+        process.exit(1);
+      }
+      break;
+    default:
+      printUsage();
```

Comment:
does hitting the default case imply there was an invalid `command` supplied? If so, may make sense to message this

Reply:
Yeah, it could be useful. I just reused the same behavior as before.

Reply:
I think we should consider using something like https://www.npmjs.com/package/make-cli for proper CLI behavior.

Reply:
[This](https://github.com/OpenZeppelin/defender-client/pull/99/commits/1edf124cb6f2c8c0e2e6b735b0180824e2f029e9) should make it more user friendly.